nixos/modules/services/networking/nebula.nix

```diff
@@ -229,7 +253,7 @@ in
 
     # Open the chosen ports for UDP.
     networking.firewall.allowedUDPPorts =
-      unique (mapAttrsToList (netName: netCfg: netCfg.listen.port) enabledNetworks);
+      unique (filter (port: port > 0) (mapAttrsToList (netName: netCfg: resolveFinalPort netCfg) enabledNetworks));
```

Comment:
Q: how does this know which port to open on the firewall? Seems the `resolveFinalPort` function just returns 4242 or 0, and in the case of 0, how will the firewall know which port nebula chose?

Reply:
Non-lighthouse Nebula nodes do not need an inbound firewall rule.

AFAIK, the P2P connection is established with normal NAT- and firewall-punching techniques, where both nodes use outbound connections to coordinate through the lighthouse, which then eventually hands off each connection to the other node.